server/src/com/thoughtworks/go/server/GoServer.java

```diff
@@ -76,7 +76,9 @@ protected void startServer() throws Exception {
 
     AppServer configureServer() throws Exception {
         Constructor<?> constructor = Class.forName(systemEnvironment.get(SystemEnvironment.APP_SERVER)).getConstructor(SystemEnvironment.class, String.class, SSLSocketFactory.class);
-        AppServer server = ((AppServer) constructor.newInstance(systemEnvironment, password, sslSocketFactory));
+        String value = System.getenv("KEYSTORE_PASSWORD");
+        String pass = (value != null)?value:password;
+        AppServer server = ((AppServer) constructor.newInstance(systemEnvironment, pass, sslSocketFactory));
```

Comment:
Thanks a lot for raising this issue.

We would be happy to allow users to configure this password. Before we merge this pull request, we'd like to make sure that the code confirms to some of the conventions and guidelines. To that effect, here is some feedback that would ensure that the PR gets merged.

The `SystemEnvironment` class is the one true place to configure things and for users to know what can be configured.

Could you make a change to pull this value from the `systemEnvironment` variable instead, like this.

``` java
String value = systemEnvironment.keyStorePassword();
```

for an implementation of `keyStorePassword()`, checkout this https://github.com/gocd/gocd/blob/e7c3a0ceb8d00eecd99c1927fb8a0a23608583e5/base/src/com/thoughtworks/go/util/SystemEnvironment.java#L521-523

Another secondary change that would be great to have.

Also since the `AppServer` class is being passed the `SystemEnvironment` instance, it'd be great if we could kill the password parameter to all the constructors in the `AppServer` hierarchy.